### DISCUSSION TITLE
Ability to set default scope

### DISCUSSION THREAD

**Discussion by spacentropy** [community/community#8890] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello!
I would like to be able to set default scope, because 99% of my searches are in our organization. 
For now I partially fixed it by changing browser's search engine autocompletion (change it to `https://cs.github.com?scopeName=<your_scope>&scope=org%3A<org_name>&q=%s`), but in cs.github.com I still have to change the scope all the time.

Thanks

**Comment by robinp-tw** [comment]
Agree. Maybe could have multiple presets - org-only and else.

 99% I also want org-only search, and get baffled when `repo:my-repo` yields `no results for /my-repo` errors.

**Comment by colinwm** [comment, marked as answer]
@spacentropy @robinp-tw great suggestion, I've heard this requested a few times. I added an option to set a default scope from the homepage:

![image](https://user-images.githubusercontent.com/78169714/151397105-4960d04b-fb2a-4401-b0ab-4fe0a8d5d23d.png)

Once you select a scope, you can set it as your default. Then, when you return to cs.github.com, it'll automatically select that scope.

**Reply by robinp-tw** [reply]
Hey @colinwm ! Thanks for the quick response! An observation on slightly split behavior though:

Indeed, now going to cs.github.com does fix the default scope, and search works well (it sticks the `?scope=org%myorg&scopeName=myorg` to the request url)

But when I do direct search from browser's search query engine support (start writing `cs` in url bar, when cs.github.com comes up, press TAB, then type search query), then the UI indicates a repo filter of `myorg` visually, but the results are still unrestricted (and the url doesn't have that restrict).

**Reply by colinwm** [reply]
@robinp-tw Great point, I didn't consider that case! I've fixed this now :smile:

**Reply by robinp-tw** [reply]
Works like a charm, thank you Colin!
